src/test/java/dev/openfeature/javasdk/EvalContextTest.java

```diff
@@ -59,4 +62,44 @@ public class EvalContextTest {
         assertEquals(2, nodeFromString.value);
         assertEquals(4, nodeFromString.left.value);
     }
+
+    @Specification(number="3.1.3", text="The evaluation context MUST support fetching the custom fields by key and " +
+            "also fetching all of the keys and values.")
+    @Test void fetch_all() {
+        EvaluationContext ec = new EvaluationContext();
+
+        ec.addStringAttribute("str", "test");
+        ec.addStringAttribute("str2", "test2");
+
+        ec.addBooleanAttribute("bool", true);
+        ec.addBooleanAttribute("bool2", false);
+
+        ec.addIntegerAttribute("int", 4);
+        ec.addIntegerAttribute("int2", 2);
+
+        ZonedDateTime dt = ZonedDateTime.now();
+        ec.addDatetimeAttribute("dt", dt);
+
+        Node<Integer> n1 = new Node<>();
+        n1.value = 4;
+        Node<Integer> n2 = new Node<>();
+        n2.value = 2;
+        n2.left = n1;
+        ec.addStructureAttribute("obj", n2);
+
+        Map<String, String> foundStr = ec.getStringAttributes();
+        assertEquals("test", foundStr.get("str"));
+        assertEquals("test2", foundStr.get("str2"));
```

Comment:
This isn't testing that we support fetching fields by key. Mind updating the tests to do this?
```suggestion
        Map<String, String> foundStr = ec.getStringAttributes();
        assertEquals(ec.getStringAttribute("str"), foundStr.get("str"));
        assertEquals(ec.getStringAttribute("str2"), foundStr.get("str2"));
```

Reply:
done